server/controllers/items/recent_public.coffee

```diff
@@ -3,29 +3,42 @@ _ = __.require 'builders', 'utils'
 error_ = __.require 'lib', 'error/error'
 items_ = __.require 'controllers', 'items/lib/items'
 bundleOwnersToItems = require './lib/bundle_owners_to_items'
+itemsQueryLimit = 100
+maxItemsPerOwner = 3
+offset = 0
 
 module.exports = (req, res)->
   { query } = req
-  assertImage = query['assert-image'] is 'true'
-  reqUserId = req.user?._id
+  { limit, lang } = query
+
+  if limit?
+    unless _.isPositiveIntegerString limit
+      return error_.bundleInvalid req, res, 'limit', limit
+
+    limit = parseInt limit
+
+  limit or= 15
 
-  itemsQueryLimit = 100
-  offset = 0
-  itemsReturnedLimit = 15
-  maxItemsPerOwner = 3
+  if lang? and not _.isLang(lang)
+    return error_.bundleInvalid req, res, 'lang', lang
+
+  lang or= 'en'
+
+  assertImage = _.parseBooleanString query['assert-image']
+  reqUserId = req.user?._id
```

Comment:
its painful to see how the typing/validation part is sooooo long (readability is smashed just to check that limit is below 100...) seriously ?! JS isnt capable of doing better that? can we have an easier convention ? i dont know any easy solution ? cant we switch to typescript ? anyway my eyes bleed can we talk before merge ? :)

Reply:
there is probably room for improvement: not knowing better, I kind of went for rewriting my own validation framework (while there are existing express solutions ([example](https://github.com/ctavan/express-validator)), but I consider controllers validation being ugly as acceptable, given it plays the role of interface between API requests (which should get nice, explicit error messages) and internal libraries (which should live in a beautiful world of pre-validated inputs, free of all the trauma of having to take input from the external world)

And I'm no expert, but I don't see how typescript would help here. But since you bring it, the first step to get there is to solve #72 ;)

But yes, you can add it to our next call agenda if this comment doesn't fully address your concern :)

Reply:
this [interface](http://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html) thing is appealing to me

```
interface Person {
    firstName: string;
    lastName: string;
}
function greeter(person: Person) {
    return "Hello, " + person.firstName + " " + person.lastName;
}
let user = { firstName: "Jane", lastName: "User" };
```